### DISCUSSION TITLE
flashlight/lib/text/decoder/LexiconSeq2SeqDecoder.cpp - Code query

### DISCUSSION THREAD

**Discussion by TzurV** [flashlight/flashlight#725] [Q&A] [answered]
**which function is called in the following line ?**:

/root/flashlight/flashlight/lib/text/decoder/LexiconSeq2SeqDecoder.cpp : line 63

    std::tie(amScores, outStates) =
        amUpdateFunc_(emissions, N, T, rawY_, rawPrevStates_, t);

The function declaration of amUpdateFunc_ I can find here:

/root/flashlight/flashlight/lib/text/decoder/LexiconSeq2SeqDecoder.h: line 21

    using AMStatePtr = std::shared_ptr<void>;
    using AMUpdateFunc = std::function<
        std::pair<std::vector<std::vector<float>>, std::vector<AMStatePtr>>(
            const float*,
            const int,
            const int,
            const std::vector<int>&,
            const std::vector<AMStatePtr>&,
            int&)>;

/root/flashlight/flashlight/lib/text/decoder/LexiconSeq2SeqDecoder.h: line 151

    AMUpdateFunc amUpdateFunc_;

**Comment by tlikhomanenko** [comment, marked as answer]
Here is the definiton of them which we pass into the decoder constructor depending on what criterion was used https://github.com/flashlight/flashlight/blob/master/flashlight/app/asr/Decode.cpp#L527-L538
